service/history/execution/context_util.go

```diff
@@ -30,6 +31,58 @@ import (
 	"github.com/uber/cadence/common/types"
 )
 
+func (c *contextImpl) emitLargeWorkflowShardIDStats() {
+	if c.shard.GetConfig().EnableShardIDMetrics() {
+		shardIDStr := strconv.Itoa(c.shard.GetShardID())
+
+		var blobSizeWarn int64
+		blobSizeGlobalWarn := c.shard.GetConfig().LargeShardHistoryBlobMetricThreshold()
+		blobSizeDomainWarn := c.shard.GetConfig().BlobSizeLimitWarn(c.GetDomainName())
+		if blobSizeGlobalWarn < blobSizeDomainWarn {
+			blobSizeWarn = int64(blobSizeGlobalWarn)
+		} else {
+			blobSizeWarn = int64(blobSizeDomainWarn)
+		}
```

Comment:
tbh it's probably easier to build a small
```go
func min64(a,b int) int64 {
  if a < b { return int64(a) }
  return int64(b)
}
```
and use it for all three.  it'd get rid of a bunch of not-intended-for-reuse temp vars too.

Reply:
might have somethiing already. Let me poke around

Reply:
yea, it might require coercing to ints or int64s or something in a few places, but we probably do have a generic-min somewhere.  though I think they're simple enough that it's fine to simply rewrite it.

Reply:
found the min64. pretty clean now i think